### DISCUSSION TITLE
PnP search getting duplicate results

### DISCUSSION THREAD

**Discussion by shirleysham** [microsoft-search/pnp-modern-search#2917] [General]
Hello,

I'm getting some duplicate results in the search results.  There are 871 records in the results:

![image](https://user-images.githubusercontent.com/121912343/232981535-9f7c2ec7-d354-4588-a237-6ad3fe93fe43.png)

But in the Sharepoint list, I only have 479 records.

![image](https://user-images.githubusercontent.com/121912343/232981733-967dac53-41b0-419a-a25d-c85d4be9a52b.png)

Here is my query template.  Also I have the trim duplicates on.  Any help would be appreciated thanks!!!!

{?((Businesstitlepeople:{searchTerms}) OR (Chinesenamepeople:{searchTerms}) OR (Departmentpeople:{searchTerms}) OR (Addressingnamepeople:{searchTerms}*) OR (Businessemailaddresspeople:{searchTerms}) AND (-"AuthorOWSUSER":{searchTerms}) AND (-"AADObjectID":{searchTerms})) (path: "https://kwmhkg.sharepoint.com/sites/Data/Lists/PeopleSearchHK" contentclass:"STS_ListItem" (Addressingnamepeople:A* OR Addressingnamepeople:B* OR Addressingnamepeople:C* OR Addressingnamepeople:D* OR Addressingnamepeople:E* OR Addressingnamepeople:F* OR Addressingnamepeople:G* OR Addressingnamepeople:H* OR Addressingnamepeople:I* OR Addressingnamepeople:J* OR Addressingnamepeople:K* OR Addressingnamepeople:L* OR Addressingnamepeople:M* OR Addressingnamepeople:N* OR Addressingnamepeople:O* OR Addressingnamepeople:P* OR Addressingnamepeople:Q* OR Addressingnamepeople:R* OR Addressingnamepeople:S* OR Addressingnamepeople:T* OR Addressingnamepeople:U* OR Addressingnamepeople:V* OR Addressingnamepeople:W* OR Addressingnamepeople:X* OR Addressingnamepeople:Y* OR Addressingnamepeople:Z* ))}

**Comment by kasperbolarsen** [comment]
Hi
If you are only looking for list items in the PeopleSearchHK  list , try to use {searchTerms} Path: "https://kwmhkg.sharepoint.com/sites/Data/Lists/PeopleSearchHK*" .

Once that works you can further limit the result

**Reply by wobba** [reply]
I would also add `contentclass:STS_ListItem_GenericList` to only return list items, and not the list or any views.

**Comment by wobba** [comment]
You cannot trust the count coming back in the search results. It's an estimate.

**Comment by shirleysham** [comment]
Hi,
We've found that the PNP search retrieves records that were deleted from
the Sharepoint lists as well.
I went into the search results web part, in debug mode, I saw duplicates in
2 records but having different paths, ie.
https://xxx.sharepoint.com/sites/Data/Lists/PeopleSearchHKClone/DispForm.aspx?ID=151
https://xxx.sharepoint.com/sites/Data/Lists/PeopleSearchHKClone/DispForm.aspx?ID=628
One of the paths had been deleted before, and one of them valid.
Is there a way to filter out the deleted records? Thanks.

**Reply by kasperbolarsen** [reply]
When an item has been deleted you should expect it to be in the search index until the search crawler picks up the change.  This will normally happen within 10 to 15 minutes,  and then you can't see it in search anymore. 
However 10-15 minutes refresh rate is not covered by a  SLA, and from time to time it can take hours or even days. In those case I'll recommend creating a service request with MS in order to let them know that there is an issue.

**Reply by koenti** [reply]
I have the same problem last week (and still) that deleted items in a SP list are still returned by the SP Search.  
It was a new SP list.  Also the deleted items are shown in de standard SP search box (at top of your site).
As temporary workaroun:
- I have updated the actuals items (so modified date is updated).
- I filter out the deleted item based on modified date.  Query example: LastModifiedTime > 2023-04-26T08:00:00Z

Removing search results with specific URLs via the search admin centre is not helping.
I probably need to raise a MS Support ticket.

**Comment by kasperbolarsen** [comment]
When Search seems to be sleeping or just unresponsive the first trick is to give it a "kick" by setting the site to be reindexed ( Site Collection setting->Search and offline availability )
